### ISSUE TITLE
Proposal to switch to another shell (electron)

### ISSUE THREAD

**Issue by ficristo** [adobe/brackets#12857]
The reasons of this proposal are:
- **maintenance**: is quite high, especially if we think about the features that are in electron but not in brackets, it can be only higher. Probably we will need to replace GYP with GN since CEF is switching to it
- **theme**: let's say the truth, the dark theme is only truly supported on macOS. On Windows 10 is not right and on Linux there isn't
- **features**: there are more features (native menu on Linux or submenu on all platforms for example)
- **platforms**: as it is personally I consider Linux simply not supported by Brackets. Yes it run on it but ATM is still on an older version of CEF and there are problems that are only ageing, electron supports all three major platforms, I'm quite sure it supports ARM and there are people looking at BSD
- **easier for contributors**: from what I saw the shell is an high impedance for contributors who wanted to work on it, slow reviews or problems on compile (Linux on master still doesn't compile)
- **easier for maintainer (especially outside Adobe)**: AFAIK @zaggino creates the installer for brackets-electron through Travis, this will mean we will able to do at least prereleases by ourself

The possible downsides are:
- **control**: if we will have a problem with the shell we will need to cooperate/rely on the electron team.

Maybe there are more downsides but ATM I cannot think anything else.

If people cannot accept to loose the dark theme they can be redirected to Dreamweaver.
Last, AFAIK, Dreamweaver doesn't use this shell, so personally I feel like Brackets is the only one to suffer the burden of this shell.

I tryed to help a bit the Linux side but the effort is simply too much time consuming for me, switching to another shell seems the best choice, and since there is already [brackets-electron](https://github.com/zaggino/brackets-electron/) seems obvious what is my choice.

**Comment by ficristo** [comment]
If I'm not mistaken @zaggino also replaced the sockets usage for node connection, which has always given some problems. This another reason to switch to the electron fork.

**Comment by zaggino** [comment]
Yes, it uses electron's IPC and also for every node connection, new node process is spawned. This way they don't interfere with each other (sync functions in one slowing down all or one bug crashing all for example) and it feels a lot faster (didn't do measurements).

**Comment by ficristo** [comment]
AFAIK the current shell scripts cannot handle patch releases, while the fork of @zaggino can.
There are issues that really deserve a patch release, think about the Ctrl-Z issue or https://github.com/adobe/brackets/pull/12145 which resulted in refreshed builds of the release.
Maybe this could be fixed in brackes-shell side, but since it is already available in the fork, why not use it? 😄
